includes/derivatives.inc

```diff
@@ -474,15 +465,63 @@ function islandora_ocr_update_datastream(AbstractObject $object, $file, $dsid, $
  * Remove the generate_ocr field if present in the RELS-EXT.
  */
 function islandora_ocr_remove_generate_ocr_rels_ext_statement(AbstractObject $object) {
-  if ($object->relationships->get(ISLANDORA_RELS_EXT_URI, 'generate_ocr', '', RELS_TYPE_PLAIN_LITERAL)) {
-    $object->relationships->remove(ISLANDORA_RELS_EXT_URI, 'generate_ocr', '', RELS_TYPE_PLAIN_LITERAL);
-  }
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  $message = filter_xss(islandora_deprecated('7.x-1.10', t('In the future, OCR-generating RELS-EXT statements should be removed using _islandora_ocr_remove_generating_rels_ext_statement().')));
+  trigger_error($message, E_USER_DEPRECATED);
+  _islandora_ocr_remove_generating_rels_ext_statement($object, ISLANDORA_OCR_GENERATE_OCR_RELS_STATEMENT);
 }
 
 /**
  * Set the generate_ocr relation in the RELS-EXT.
  */
 function islandora_ocr_set_generate_ocr_rels_ext_statement(AbstractObject $object, $generate_ocr = TRUE) {
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  $message = filter_xss(islandora_deprecated('7.x-1.10', t('In the future, OCR-generating RELS-EXT statements should be set together using islandora_ocr_set_generating_rels_ext_statements().')));
+  trigger_error($message, E_USER_DEPRECATED);
   $generate_ocr = $generate_ocr ? 'TRUE' : 'FALSE';
-  $object->relationships->add(ISLANDORA_RELS_EXT_URI, 'generate_ocr', $generate_ocr, RELS_TYPE_PLAIN_LITERAL);
+  $object->relationships->add(ISLANDORA_RELS_EXT_URI, ISLANDORA_OCR_GENERATE_OCR_RELS_STATEMENT, $generate_ocr, RELS_TYPE_PLAIN_LITERAL);
+}
+
+/**
+ * Helper to remove OCR/HOCR generation RELS-EXT statements.
+ *
+ * @param AbstractObject $object
+ *   The object to remove relationships from.
+ * @param string $flag
+ *   The flag to remove.
+ */
+function _islandora_ocr_remove_generating_rels_ext_statement(AbstractObject $object, $flag) {
```

Comment:
Why is this an internal "_" function but the (new) setter is a global available one? I

Reply:
This is consistent with how they're used. This function (`_islandora_ocr_remove_generating_rels_ext_statement()`) is only used (or intended for, in any case) inside of this module... The setter (`islandora_ocr_set_generating_rels_ext_statements()`) is called externally, in all the related pull requests:
* [`islandora_paged_content`](https://github.com/Islandora/islandora_paged_content/pull/131/files#diff-513c241a5c2f213d72b035bc1baa0743R805)
* [`islandora_solution_pack_book`](https://github.com/Islandora/islandora_solution_pack_book/pull/196/files#diff-25b6cf4bd16ae6912ebb116b013bc6a5R286)
* [`islandora_book_batch`](https://github.com/Islandora/islandora_book_batch/pull/68/files#diff-9d74701071b57981e86c31d3ea01aad0R480); and,
* [`islandora_newspaper_batch`](https://github.com/Islandora/islandora_newspaper_batch/pull/21/files#diff-bb0eb85e45687085a3b6815a5c75a4a2R496)

... Is there an issue with this?